flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/gateway/local/LocalExecutorITCase.java

```diff
@@ -222,7 +222,7 @@ public void testListUserDefinedFunctions() throws Exception {
 
 		final List<String> actualTables = executor.listUserDefinedFunctions(session);
 
-		final List<String> expectedTables = Arrays.asList("aggregateUDF", "tableUDF", "scalarUDF");
+		final List<String> expectedTables = Arrays.asList("aggregateUDF", "scalarUDF", "tableUDF");
```

Comment:
I think either we should mention in the JavaDoc that the returned list is sorted, or the test should verify with a set?

Reply:
I think I should move the sort logic to CliClient#callShowFunctions(), since neither TableEnv#listFunctions() nor FunctionCatalog#getFunctions() care about the sorted order. Sorted order is only for convenience of displaying to users in SQL CLI.

Reply:
Personally I don't think we need to return a sorted array. SQL-client can sort the result set and present to the console.